### PR TITLE
72531-NEW Delete/Redistribute Install with Native Helm Chart

### PR DESCRIPTION
**Pull request by jonquil2002**
SC: https://app.shortcut.com/replicated/story/72531/delete-consolidate-topic-installing-with-native-helm

This branch replaces the original 72531 branch.

Previews:

- https://deploy-preview-1042--replicated-docs.netlify.app/vendor/helm-native-helm-install-order
- https://deploy-preview-1042--replicated-docs.netlify.app/vendor/helm-overview#using-the-app-manager
- https://deploy-preview-1042--replicated-docs.netlify.app/vendor/helm-release

### CODE REVIEW

File: docs/vendor/helm-native-helm-install-order.md [hunk 1]
```diff
@@ -70,6 +70,20 @@ For more information about using Helm dependencies, see [Chart Dependencies](htt
...
+* pre-install - executes after resources are rendered but before any resources are installed.
+* post-install - executes after resources are installed.
+* pre-upgrade - executes after resources are rendered but before any resources are upgraded.
+* post-upgrade - executes after resources are upgraded.
+
+The following hooks can be used but no actions will be taken by Replicated:
+* pre-rollback - executes after resources are rendered but before any resources are rolled back.
+* post-rollback - executes after resources are rolled back.
+* pre-delete - executes before any resources are deleted.
+* post-delete - executes after resources are deleted.
```
Comment (ajp-io) [inline docs/vendor/helm-native-helm-install-order.md:87]: What does this mean? You can specify these and it won't break anything but it won't do anything either? Or just that Replicated doesn't specifically have to do anything for these? I assume the former, but it's not clear to me since we say they "can be used".

File: docs/vendor/helm-overview.md [hunk 2]
```diff
@@ -2,13 +2,13 @@ import NativeHelmLimitations from "../partials/helm/_native-helm-limitations.mdx
...
 
 Replicated supports delivering an enterprise application as Helm charts, or including Helm charts as components of an application. An application can use more than one Helm chart, and can use more than a single instance of any Helm chart.
 
-To package an application with Helm, start by adding an existing Helm chart to a release in the Replicated vendor portal. For information about how to create a new release from an existing Helm chart, see [Creating Releases with Helm Charts](helm-release).
+To deploy an application with Helm, start by adding an existing Helm chart to a release in the Replicated vendor portal. For information about how to create a new release from an existing Helm chart, see [Creating Releases with Helm Charts](helm-release).
```
Comment (ajp-io) [inline docs/vendor/helm-overview.md:11]: WDYT? "deploy an application" doesn't sound right since the chart is just part of the application.
```suggestion
To deploy a Helm chart, start by adding the Helm chart to a release in the Replicated vendor portal. For information about how to create a new release from an existing Helm chart, see [Creating Releases with Helm Charts](helm-release).
```

File: docs/vendor/helm-overview.md [hunk 3]
```diff
@@ -18,19 +18,24 @@ When you distribute an application packaged with Helm, your customers can instal
 
 Users can install an application packaged with Helm charts using the app manager on an existing cluster or on a cluster provisioned by the Kubernetes installer. When installing with the app manager, users can either use the Replicated admin console UI or the kots CLI.
 
-The app manager installs applications packaged with Helm charts using either the "Native Helm" or "Replicated KOTS" deployment method. You specify the deployment method in the Replicated HelmChart custom resource manifest file with the `useHelmInstall` flag. For more information, see [useHelmInstall](/reference/custom-resource-helmchart#usehelminstall) in _HelmChart_.
+The app manager installs applications packaged with Helm charts using either the **Native Helm** or **Replicated Helm** deployment method. You specify the deployment method in the Replicated HelmChart custom resource manifest file with the `useHelmInstall` flag. For more information, see [useHelmInstall](/reference/custom-resource-helmchart#usehelminstall) in _HelmChart_.
```
Comment (ajp-io) [inline docs/vendor/helm-overview.md:21]: Same thoughts as above.
```suggestion
The app manager installs Helm charts using either the **Native Helm** or **Replicated Helm** deployment method. You specify the deployment method in the Replicated HelmChart custom resource manifest file with the `useHelmInstall` flag. For more information, see [useHelmInstall](/reference/custom-resource-helmchart#usehelminstall) in _HelmChart_.
```

File: docs/vendor/helm-overview.md [hunk 4]
```diff
@@ -18,19 +18,24 @@ When you distribute an application packaged with Helm, your customers can instal
...
 
 * **Native Helm (Recommended)**: The app manager uses the Helm binary to install and manage the lifecycle of the chart resources that are part of the application. This is the preferred method because it supports more features of Helm, such as hooks and weights.
 
-For more information, see [Native Helm](helm-processing#native-helm) in _How the App Manager Processes Helm Charts_. See also [Enabling and Using Native Helm Charts](helm-installing-native-helm#enabling-and-using-native-helm-charts) in _Installing with Native Helm_.
+  For more information, see [Native Helm](helm-processing#native-helm) in _How the App Manager Processes Helm Charts_. See also [Creating Releases with Helm Charts](helm-release).
 
-* **Replicated KOTS**: The app manager renders the Helm templates and deploys them as standard Kubernetes manifests using `kubectl apply`. The app manager manages the lifecycle of the resources.
+  :::note
+  Migrating existing installations to the native Helm workflow is not supported. However, new Helm charts within an existing application can use the native Helm workflow and the features that come with it.
```
Comment (ajp-io) [inline docs/vendor/helm-overview.md:30]: I'm just toying around, but this makes a bit more sense to me.
```suggestion
  Charts installed with Replicated Helm cannot be migrated to use Native Helm. However, Native Helm can always be used for charts that are newly added to a release, even if other charts in the release use Replicated Helm.
```

File: docs/vendor/helm-release.mdx [hunk 5]
```diff
@@ -1,21 +1,22 @@
...
 # Creating Releases with Helm Charts
 
 This topic describes how to add a Helm chart to a release in the Replicated vendor portal or the replicated CLI. Adding a Helm chart to a release is useful if you want to distribute a Kubernetes application with Replicated that is already packaged using Helm.
 
-This procedure applies for Replicated Helm installations and native Helm installations. For more information, see [Helm Overview](helm-overview).
-
 ## About Releasing with Helm Charts
 
-You can add one or more Helm charts to a release in the vendor portal by uploading each Helm chart as a `.tgz` package. This allows you to use Replicated to distribute applications that are packaged with Helm.
+You can add one or more Helm charts to a release in the vendor portal by uploading each Helm chart as a `.tgz` package. This allows you to use Replicated to distribute applications that are packaged with Helm. The procedures in this topic apply to native Helm installations (recommended) and Replicated Helm installations. For more information, see [About Deploying with Helm](helm-overview).
```
Comment (ajp-io) [inline docs/vendor/helm-release.mdx:13]: I think .tgz file is more standard wording.

What do you think about the "in part or in whole" part, since it doesn't have to be all or nothing?
```suggestion
You can add one or more Helm charts to a release in the vendor portal by uploading each Helm chart as a `.tgz` file. This allows you to use Replicated to distribute applications that are packaged in part or in whole with Helm. The procedures in this topic apply to native Helm installations (recommended) and Replicated Helm installations. For more information, see [About Deploying with Helm](helm-overview).
```

File: docs/vendor/helm-native-helm-install-order.md (hunk 1 above)
Comment (ajp-io) [reply docs/vendor/helm-native-helm-install-order.md:87]: I guess this wording exists today and we aren't modifying it, just moving it. But I don't think it's clear what this means. If they don't do anything, we should say that instead of "the following hooks can be used". They can't really be used if they don't do anything, which is my inference here.

But it sounds like Craig isn't sure either, so this might take investigation.

File: docs/vendor/helm-release.mdx [hunk 6]
```diff
@@ -1,21 +1,22 @@
...
+
+When you add a Helm chart to a release, Replicated adds a copy of the `Chart.yaml` file and the `values.yaml` file from the Helm chart to the release.
```
Comment (cbodonnell) [inline docs/vendor/helm-release.mdx:15]: We don't really _add copies_ of the `Chart.yaml` file and the `values.yaml` file, but rather _display copies_.  I'm assuming it's for reference purposes to help the vendor fill out the HelmChart custom resource.

File: docs/vendor/helm-overview.md [hunk 7]
```diff
@@ -47,10 +52,10 @@ This section lists the limitations for packaging an application with Helm charts
 
 There are different limitations depending on if your customers install and manage the application with the app manager, or if they use the helm CLI directly:
 
-* [Replicated KOTS and Native Helm Limitations](#replicated-helm-limitations)
+* [Replicated Helm and Native Helm Limitations](#replicated-helm-limitations)
 * [helm CLI Install Limitations](#helm-cli-limitations)
 
-### Native Helm and Replicated KOTS Limitations {#replicated-helm-limitations}
+### Native Helm and Replicated Helm Limitations {#replicated-helm-limitations}
```
[the remark this answers is not in the document]
Comment (jonquil2002) [reply docs/vendor/helm-overview.md:58]: Thanks!

File: docs/vendor/helm-release.mdx [hunk 8]
```diff
@@ -162,7 +171,7 @@ To create and promote a new release:
 
 1. Create a HelmChart custom resource manifest file in the same directory where you copied the `.tgz` file. Name the HelmChart manifest file `CHART_NAME.yaml`. For example, `postgresql.yaml`.
 
-1. Copy and paste the following YAML into the HelmChart manifest:
+1. Copy and paste the following YAML into the HelmChart manifest. The `useHelmInstall` flag is set to `true`, specifying that the release uses the recommended native Helm installation. For more information, see [HelmChart](../reference/custom-resource-helmchart) in the _Custom Resources_ section.
```
[the remark this answers is not in the document]
Comment (jonquil2002) [reply docs/vendor/helm-release.mdx:174]: "native" should only be capitalized in headings and at the beginning of a sentence. Since the actual product name is Helm and native is a descriptor, not a proper noun, we need to go with lowercase in general. I'll do a search for non-compliant usage. Thanks for catching that and inquiring. If you disagree, let us know. :)

File: docs/vendor/helm-release.mdx [hunk 9]
```diff
@@ -1,21 +1,22 @@
...
-
 ## About Releasing with Helm Charts
 
-You can add one or more Helm charts to a release in the vendor portal by uploading each Helm chart as a `.tgz` package. This allows you to use Replicated to distribute applications that are packaged with Helm.
+You can add one or more Helm charts to a release in the vendor portal by uploading each Helm chart as a `.tgz` package. This allows you to use Replicated to distribute applications that are packaged with Helm. The procedures in this topic apply to native Helm installations (recommended) and Replicated Helm installations. For more information, see [About Deploying with Helm](helm-overview).
+
+When you add a Helm chart to a release, Replicated adds a copy of the `Chart.yaml` file and the `values.yaml` file from the Helm chart to the release. 
 
-When you add a Helm chart to a release, Replicated adds a copy of the `Chart.yaml` file and the `values.yaml` file from the Helm chart to the release.
+You must also have a HelmChart custom resource manifest file for each Helm chart that you add to a release. By default, Replicated adds the HelmChart custom resource for you, which you use to configure the Helm chart with your application.
```
[the remark this answers is not in the document]
Comment (jonquil2002) [reply docs/vendor/helm-release.mdx:17]: Yes and yes, incorporated both of your suggestions. (Added them manually.)

File: docs/vendor/helm-native-helm-install-order.md (hunk 1 above)
Comment (jonquil2002) [reply docs/vendor/helm-native-helm-install-order.md:87]: @salah is going to investigate this and we can resolve it in a separate PR.

Comment (cbodonnell) [reply docs/vendor/helm-native-helm-install-order.md:87]: @jonquil2002 I was able to confirm that `pre-delete` and `post-delete` hooks are invoked when a helm chart is uninstalled in KOTS (for example if you remove it from your application).  The rollback hooks would not be invoked since we don't ever run `helm rollback`.

File: docs/vendor/helm-release.mdx [hunk 10]
```diff
@@ -51,6 +50,14 @@ For example, the following screenshot shows how a Postgres Helm chart displays i
 
 ![Postgres Helm Chart](/images/postgres-helm-chart.png)
 
+[View a larger image](/images/postgres-helm-chart.png)
+
+## About Updating Existing Applications
```
Comment (paigecalvert) [inline docs/vendor/helm-release.mdx:55]: Similar to the note above that I commented on, I love that you added this description.

This heading was a little confusing for me; it sounded like it might be about an end user updating a helm app, or that there are special steps you have to take as a vendor to make changes to an app that uses helm charts. The content doesn't really mention updating an app, per se

What do you think about something like "About Updating Replicated Helm Releases" or "About Adding New Charts to Releases"?

Alternatively, I also noticed that this content is similar to existing Limitations about native Helm: https://docs.replicated.com/vendor/helm-overview#replicated-helm-limitations. It also looks similar to the note you added above: "Charts installed with Replicated Helm cannot be migrated to use native Helm. However, native Helm can always be used for charts that are newly added to a release, even if other charts in the release use Replicated Helm."

Specifically, "Only supported for new installations" and "Not supported on existing charts deployed on existing applications." Maybe this information would fit well as part of those limitations instead? As is, the limitations are super lightweight and don't give a lot of good info about what this means. Your description is so much better

TLDR; looks like there's an opportunity to condense info about migrating from Replicated Helm to native Helm under limitations perhaps

**Review by paigecalvert — APPROVED** [review]

File: docs/vendor/helm-release.mdx (hunk 10 above)
Comment (jonquil2002) [reply docs/vendor/helm-release.mdx:55]: Good idea about the heading, I changed it.

Re: the content overlap, that's definitely something I've been running into a lot with Helm. It's scattershot content and I've been condensing a lot of it. But now it's getting way beyond the scope of my SC. Also, with the changes coming from your PR, I think it's better to wait until it's all merged and then re-evaluate things. It just seems kinda weird that Helm is in two sections and we may want to reconsider that holistically. I'll open an SC for it, if there isn't one already.

File: docs/vendor/helm-native-helm-install-order.md (hunk 1 above)
Comment (jonquil2002) [reply docs/vendor/helm-native-helm-install-order.md:87]: Thank you @cbodonnell !
